src/app/api/Api.ts

```diff
@@ -274,9 +281,10 @@ export interface API {
 
     // Sheriff Shifts
     getShifts(): Promise<Shift[]>;
-    updateShift(shiftToUpdate: Partial<Shift>): Promise<Shift>;
+    updateSelectedShifts(shiftIds: IdType[], shiftUpdates: ShiftUpdates): Promise<Shift[]>;
+    updateShift(updatedShift: Partial<Shift>): Promise<Shift>;
     createShift(newShift: Partial<Shift>): Promise<Shift>;
-    deleteShift(shiftId: IdType): Promise<void>;
```

Comment:
so changes to signatures in api's generally break things as you most likely found out.  In these situations I would generally avoid changing the signature of something already in there and instead create new methods for the new signatures.  For instance:

*OPTION 1*
```ts
deleteShift(shiftId:IdType): Promise<void>{
  return this.deleteShifts([shiftId]);
}

deleteShifts(shiftIds:IdType[]):Promise<void>{
 //  ... actually do delete
}
```

or.. I believe you could have also done something like this:

*OPTION 2*
```ts
    deleteShift(...ids:string[]){
      // .. do delete
    }
// Which can be called like this:
        this.deleteShift('someString');
        this.delete('someid','anotherid');
        this.deleteShift(...['many','ids']);
```

But you can see that with the option 2 that we can call the method the same way as we originally did but can spread arrays of values into it if needed which I think is a nice way to handle this type of thing.

Reply:
Makes sense! Will keep that in mind for next time - or should we updated the code now?